### PR TITLE
[Delivers #168789986] Migrate Interactions category to use aws-sdk v3 clients

### PR DESCRIPTION
**Pull request by jordanranz**
Related to #3365

Description of changes:
This PR upgrades the aws-sdk v2 clients used in the Interactions category to aws-sdk V3 client. This helps achieve modularization at the AWS clients level in Interactions.

[Delivers #168789986]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/interactions/__tests__/Interactions-unit-test.ts [hunk 1]
```diff
@@ -1,66 +1,70 @@
-jest.mock('aws-sdk/clients/lexruntime', () => {
-	const LexRuntime = () => {};
-	LexRuntime.prototype.postText = (params, callback) => {
-		if (params.inputText === 'done') {
-			callback(null, {
-				message: 'echo:' + params.inputText,
+import { LexRuntimeServiceClient } from '@aws-sdk/client-lex-runtime-service-browser/LexRuntimeServiceClient';
+import Interactions from '../src/Interactions';
+import { findInterfacesAddedToObjectTypes } from 'graphql/utilities/findBreakingChanges';
```
Comment (elorzafe) [inline packages/interactions/__tests__/Interactions-unit-test.ts]: Why is this needed? I know I added lol, but it seems it shouldn't be there

File: packages/interactions/src/Providers/AWSLexProvider.ts [hunk 2]
```diff
@@ -12,153 +12,151 @@
...
-	InteractionsOptions,
-	InteractionsMessage,
-	InteractionsResponse,
-} from '../types';
-import * as LexRuntime from 'aws-sdk/clients/lexruntime';
+import { InteractionsOptions, InteractionsMessage } from '../types';
+import { LexRuntimeServiceClient } from '@aws-sdk/client-lex-runtime-service-browser/LexRuntimeServiceClient';
+import { PostTextCommand } from '@aws-sdk/client-lex-runtime-service-browser/commands/PostTextCommand';
+import { PostContentCommand } from '@aws-sdk/client-lex-runtime-service-browser/commands/PostContentCommand';
 import { ConsoleLogger as Logger, AWS, Credentials } from '@aws-amplify/core';
```
Comment (elorzafe) [inline packages/interactions/src/Providers/AWSLexProvider.ts]: `AWS` object won't be needed here

File: packages/interactions/src/Providers/AWSLexProvider.ts [hunk 3]
```diff
@@ -12,153 +12,151 @@
...
+        if (!this._config[botname]) {
+            return Promise.reject('Bot ' + botname + ' does not exist');
+        }
+        const credentials = await Credentials.get();
+        if (!credentials) {
+            return Promise.reject('No credentials');
+        }
+        AWS.config.update({
+            credentials,
+        });
```
Comment (elorzafe) [inline packages/interactions/src/Providers/AWSLexProvider.ts]: This is not needed

File: packages/interactions/src/Providers/AWSLexProvider.ts [hunk 4]
```diff
@@ -12,153 +12,151 @@
...
+                    userId: credentials.identityId,
+                    accept: 'audio/mpeg',
+                };
+            }
+
+            logger.debug('postContent to lex', message);
+
+            try {
+                const postContentCommand = new PostContentCommand(params);
+                const data = this.lexRuntimeServiceClient.send(postContentCommand);
```
Comment (elorzafe) [inline packages/interactions/src/Providers/AWSLexProvider.ts]: Is this synchronous?

File: packages/interactions/src/Providers/AWSLexProvider.ts [hunk 5]
```diff
@@ -12,153 +12,151 @@
...
+                this.responseCallback(data, botname);
```
Comment (elorzafe) [inline packages/interactions/src/Providers/AWSLexProvider.ts]: `this.responseCallback` is asynchronous, completion callback can be called after the response is sent

File: packages/interactions/__tests__/Interactions-unit-test.ts (hunk 1 above)
Comment (jordanranz) [reply packages/interactions/__tests__/Interactions-unit-test.ts]: I do not believe it is needed, I can remove it.

File: packages/interactions/src/Providers/AWSLexProvider.ts (hunk 2 above)
Comment (jordanranz) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: Awesome, I can remove.

File: packages/interactions/src/Providers/AWSLexProvider.ts (hunk 4 above)
Comment (jordanranz) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: Good catch

File: packages/interactions/src/Providers/AWSLexProvider.ts (hunk 5 above)
Comment (jordanranz) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: what do you mean by "completion callback" here?

File: packages/interactions/src/Providers/AWSLexProvider.ts (hunk 3 above)
Comment (jordanranz) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: Removing this

File: packages/interactions/src/Providers/AWSLexProvider.ts [hunk 6]
```diff
@@ -12,153 +12,148 @@
...
+                userId: credentials.identityId,
+            };
+
+            logger.debug('postText to lex', message);
+
+            try {
+                const postTextCommand = new PostTextCommand(params);
+                const data = await this.lexRuntimeServiceClient.send(postTextCommand);
+                this.responseCallback(data, botname);
+                return data;
```
Comment (pravgupt) [inline packages/interactions/src/Providers/AWSLexProvider.ts]: Shouldn't we return the response from the responseCallBack() as was before in case the callback changes/updates the data or conditionally return the response? Also we should rename the method as it isn't technically callback anymore.

File: packages/interactions/src/Providers/AWSLexProvider.ts [hunk 7]
```diff
@@ -12,153 +12,148 @@
...
-					typeof this._config[botname].onComplete === 'function'
-				) {
-					setTimeout(
-						() => this._config[botname].onComplete(null, { slots: data.slots }),
-						0
-					);
-				}
-			}
-
-			res(data);
```
Comment (pravgupt) [inline packages/interactions/src/Providers/AWSLexProvider.ts]: Is it logically same to move the resolve from here in the middle of this method to skip it to the end?

Comment (jordanranz) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: This will logically be the same.

File: packages/interactions/src/Providers/AWSLexProvider.ts (hunk 6 above)
Comment (jordanranz) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: I can change the name of the function here. I mainly left it around to preserve the content but I think it is fair to change the naming of it to better describe the functionality.

File: packages/interactions/src/Providers/AWSLexProvider.ts (hunk 5 above)
Comment (jordanranz) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: Spoke offline about this. Seems the responseCallback is just doing some error logging to the chatbot. I will change the name of it to better describe the functionality.

File: packages/interactions/src/Providers/AWSLexProvider.ts (hunk 6 above)
Comment (elorzafe) [reply packages/interactions/src/Providers/AWSLexProvider.ts]: `responseCallBack` is invoking onComplete callbacks that are configured. I think `reportBotStatus` is accurate
